IT subtests `manifest` and `repository_files` fail due to lack of files

https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/2673

```
======================================================================
FAIL: test_indexing (integration_test.IndexingIntegrationTest) [manifest] (catalog='anvil-it', format=None)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 378, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 559, in _test_manifest
    validator(catalog, response)
  File "/builds/ucsc/azul/test/integration_test.py", line 684, in _check_manifest
    self.__check_manifest(BytesIO(response), 'bundle_uuid')
  File "/builds/ucsc/azul/test/integration_test.py", line 760, in __check_manifest
    self.assertGreater(len(rows), 0)
AssertionError: 0 not greater than 0
======================================================================
FAIL: test_indexing (integration_test.IndexingIntegrationTest) [manifest] (catalog='anvil-it', format=<ManifestFormat.compact: 'compact'>)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 378, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 559, in _test_manifest
    validator(catalog, response)
  File "/builds/ucsc/azul/test/integration_test.py", line 684, in _check_manifest
    self.__check_manifest(BytesIO(response), 'bundle_uuid')
  File "/builds/ucsc/azul/test/integration_test.py", line 760, in __check_manifest
    self.assertGreater(len(rows), 0)
AssertionError: 0 not greater than 0
======================================================================
FAIL: test_indexing (integration_test.IndexingIntegrationTest) [repository_files] (catalog='anvil-it')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 378, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 788, in _test_repository_files
    file = self._get_one_file(catalog)
  File "/builds/ucsc/azul/test/integration_test.py", line 581, in _get_one_file
    self.fail('No files found')
AssertionError: No files found
```


There seem to be many bundles without any files in the 1000G snapshot, so the integration subtests that rely on presence of files in the index fail frequently, depending on the random seed chosen.

<!--
Add your description here. For suspected bugs, the steps required to reproduce
the problem, as well as the expected and actual outcome should be included in
the description. Solutions should be proposed in a  separate comment, not the
issue description. Please do not modify or delete the checklist below. The
checklist should remain at the end of the issue description.
-->

- [ ] Security design review completed; the Resolution of this issue does **not** …
  - [ ] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [ ] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [ ] … affect the permissions of external users like access to
    - TDR snapshots
  - [ ] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [ ] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [ ] … affect monitoring of the system
  - [ ] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [ ] … add an interface that exposes sensitive or confidential data at the security boundary
  - [ ] … affect the encryption of data at rest
  - [ ] … require persistence of sensitive or confidential data that might require encryption at rest
  - [ ] … require unencrypted transmission of data within the security boundary
  - [ ] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [ ] Documentation on any unchecked boxes is provided in comments below

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -335,7 +335,8 @@ class IntegrationTestCase(AzulTestCase, metaclass=ABCMeta):
                 map(str, self.managed_access_sources_by_catalog[catalog])
             )
             index = first(
-                i for i, source in enumerate(sources)
+                i
+                for i, source in enumerate(sources)
                 if source not in managed_access_sources
             )
             sources[0], sources[index] = sources[index], sources[0]

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -205,13 +205,7 @@ class IntegrationTestCase(AzulTestCase, metaclass=ABCMeta):
         super().setUp()
         # All random operations should be made using this seed so that test
         # results are deterministically reproducible
-        self.random_seed = (
-            # FIXME: Unpin the seed once underlying issue is fixed
-            #        https://github.com/DataBiosphere/azul/issues/5168
-            6634795309975096822
-            if config.deployment_stage == 'anvilprod' else
-            randint(0, sys.maxsize)
-        )
+        self.random_seed = randint(0, sys.maxsize)
         self.random = Random(self.random_seed)
         log.info('Using random seed %r', self.random_seed)
 
@@ -332,7 +326,8 @@ class IntegrationTestCase(AzulTestCase, metaclass=ABCMeta):
         self.random.shuffle(sources)
         if public_1st:
             managed_access_sources = frozenset(
-                map(str, self.managed_access_sources_by_catalog[catalog])
+                str(source.spec)
+                for source in self.managed_access_sources_by_catalog[catalog]
             )
             index = first(
                 i

